src/rules/patternDetector.js

```diff
@@ -1,68 +1,192 @@
 "use strict";
 const fs = require("fs");
-const ioUtils = require("../lib/ioUtils");
-const appRoot = require("app-root-path");
-const script = appRoot.path + "/deps/py/pattern_matcher.py";
-const patternMatcherResults =
-  appRoot.path + "/deps/py/pattern-matcher-results.json";
 const ruleType = "FMNCP";
-const exec = require("child_process").exec;
-
 let logger = require("bunyan");
-let log = logger.createLogger({ name: "std-gov-oos" });
+let log = logger.createLogger({ name: "standardly" });
 
 /**
- * Runs python script to find pattern matches.
- * @param {*} repo
- * @returns {Promise}
+ * Loads the list of patterns from the input file and parses them, returning the
+ * pattern if it matches the rule type and undefined if not.
+ *
+ * @param {[String, Buffer, URL]} inputFilepath
+ * @param {String} ruleType
+ *
+ * @returns {Array} pattern list
  */
-function runPy(repo, rules, excludeDirs) {
-    return new Promise(resolve => {
-        try {
-            exec("python " + script + " -i " + rules + " -d " + repo + " -o " + patternMatcherResults +
-          " -k " + ruleType + " -e " + excludeDirs,
-            { stdio: "inherit" },
-            (error, stderr) => {
-                if (error) {
-                    log.error(stderr);
-                    resolve(false);
+function loadRuleJSON(inputFilepath, ruleType) {
```

Comment:
instead of pass the rulesfile, the ruleset can be passed from the testlet here: https://github.com/jlu17/standardly/blob/py-and-js/src/rules/PatternNonExistenceTestlet.js#L27